### PR TITLE
[CMP] put Prebid behind consent everywhere

### PR DESCRIPTION
**Pull request by mxdvl**
## What does this change?

Prebid now only runs if consent is given.

## Does this change need to be reproduced in dotcom-rendering ?

- [X] No
- [ ] Yes (please indicate your plans for DCR Implementation)

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [ ] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [ ] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [ ] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [X] Locally
- [x] On CODE (optional)

### CODE REVIEW

**Review by sndrs — APPROVED** [review]
🤺

**Review by ioanna0 — APPROVED** [review]

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-prebid.js [hunk 1]
```diff
@@ -30,10 +33,8 @@ const loadPrebid: () => void = () => {
 
 const setupPrebid: () => Promise<void> = () => {
     onConsentChange(state => {
-        // Only TCFv2 mode can prevent running Prebid
-        const canRun: boolean = state.tcfv2
-            ? getConsentFor('prebid', state)
-            : true;
+        // TCFv2 and AUS mode can prevent running Prebid
```
Comment (ioanna0) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-prebid.js]:
```suggestion
        // All CMP modes (TCFv2, CCPA and AUS) can prevent running Prebid
```

File: static/src/javascripts/projects/commercial/modules/dfp/prepare-prebid.spec.js [hunk 2]
```diff
@@ -150,6 +171,46 @@ describe('init', () => {
         expect(prebid.initialise).not.toBeCalled();
     });
 
+    it('should initialise Prebid in CCPA if doNotSell is false', async () => {
+        dfpEnv.hbImpl = { prebid: true, a9: false };
+        commercialFeatures.dfpAdvertising = true;
+        commercialFeatures.adFree = false;
+        onConsentChange.mockImplementation(ccpaWithConsentMock);
+        getConsentFor.mockReturnValue(true); // TODO: Why do we need to mock this?
```
Comment (mxdvl) [inline static/src/javascripts/projects/commercial/modules/dfp/prepare-prebid.spec.js:179]: I wish we didn’t mock this…

Comment (ioanna0) [reply static/src/javascripts/projects/commercial/modules/dfp/prepare-prebid.spec.js:179]: last time to use the real one it was giving ES2020 errors like this one: https://chat.google.com/room/AAAAWwBdSMs/kxMrGKQfdiY

**Pull request merged**